syntax = "proto3";
package user;

message User {
    int32 id = 1;
    string user_name =2;
    string email = 3;
    string password = 4;
}

message RegisterUserRequest {
    string user_name =1 ;
    string email = 2 ;
    string password = 3;
}

message RegisterUserResponse{
    string message = 1 ;
}

message GetUserRequest{
    int32 parent_id = 1;
}

message GetUserResponse {
    User user = 1;
}

message LoginUserRequest{
    string user_name =1;
    string password = 2;
}

message LoginUserResponse{
    string access_token = 1;
}

service UserService{
    rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse) {}
    rpc GetUser (GetUserRequest) returns (GetUserResponse) {}
    rpc LoginUser (LoginUserRequest) returns (LoginUserResponse) {}
}

